system-test/spanner.ts

```diff
@@ -4393,6 +4396,37 @@ describe('Spanner', () => {
       });
     });
   });
+
+  describe('Resource Based Routing', () => {
+    const database = instance.database(generateName('database'));
+    const table = database.table('ResourceBasedRoutingTable');
+
+    const schema = `
+      CREATE TABLE UserTable (
+        Key STRING(MAX) NOT NULL,
+        Name STRING(MAX)
+      ) PRIMARY KEY (Key)
+    `;
+
+    before(async () => {
+      await onPromiseOperationComplete(await database.create());
+      await onPromiseOperationComplete(await table.create(schema));
+    });
+    beforeEach(() => {
+      spanner.options.enableResourceBasedRouting = true;
+    });
+    afterEach(() => {
+      spanner.options.enableResourceBasedRouting = false;
+    });
+
+    it('should list the session while resource based routing is enabled', done => {
+      database.getSessions((err, session) => {
+        assert.ifError(err);
+        assert.equal(session!.length, 0);
+        done();
+      });
```

Comment:
In the spec we also wanted a test for user overriding the endpoint in the client options.

User-specified endpoint config

1. Set up the client to point at an instance-specific endpoint
2. Start the client / make a Spanner api call
3. call Spanner api (e.g. ListSessions). Expected to call the configured instance-config endpoint (GetInstance response should be empty, resulting in no override).

Reply:
@skuruppu Added above system test case.